src/lib/api-client.js

```diff
@@ -37,14 +38,31 @@ function post(endpoint, data) {
     });
 }
 
+export function checkInItem(studentId, itemAddress){
+    post('checkin', {
+        studentId,
+        itemAddress
+    }).then(data => {
+        Dispatcher.handleAction('CHECKIN_SUCCESS', {
+            itemAddress: data.itemAddress
+        });
+    }).catch(data => {
+        Dispatcher.handleAction('ERROR', {
+            error: data.error
+        });
+    });
+}
+
 export function checkOutItems(studentId, itemAddresses){
     post('checkout', {
         studentId,
         itemAddresses
     }).then(() => {
         Dispatcher.handleAction('CHECKOUT_SUCCESS');
-    }).catch(() => {
-        Dispatcher.handleAction('CHECKOUT_FAILED');
+    }).catch(data => {
```

Comment:
If the `post` fails, the error message will be the parameter for `catch`. `data` would be a string, and the `data.error` below would be undefined.